engine/graphics/src/opengl/graphics_opengl.cpp

```diff
@@ -1620,12 +1622,21 @@ static uintptr_t GetExtProcAddress(const char* name, const char* extension_name,
 
     static void BindVertexDeclarationProgram(HContext context, HVertexDeclaration vertex_declaration, HProgram program)
     {
-
+        OpenGLProgram* program_ptr = (OpenGLProgram*) program;
         uint32_t n = vertex_declaration->m_StreamCount;
         VertexDeclaration::Stream* streams = &vertex_declaration->m_Streams[0];
         for (uint32_t i=0; i < n; i++)
         {
-            GLint location = glGetAttribLocation(program, streams[i].m_Name);
+            int32_t location = -1;
+            for (int j = 0; j < program_ptr->m_Attributes.Size(); ++j)
+            {
+                if (program_ptr->m_Attributes[j].m_NameHash == streams[i].m_NameHash)
+                {
+                    location = program_ptr->m_Attributes[j].m_Location;
+                    break;
+                }
+            }
```

Comment:
The location for every attribute gets cached in the program now, which is one less GL call in the draw setup!